### PR TITLE
EVG-16501: clean up error message style

### PR DESCRIPTION
**Pull request by Kimchelly**
Jira: https://jira.mongodb.org/browse/EVG-16501

* Clean up error message style.
* Fix a bug in interval queue group operation which would attempt to use a nil queue with the interval operations.

### CODE REVIEW

File: pool/rate_limiting_average.go [hunk 1]
```diff
@@ -23,26 +22,14 @@ import (
...
-	}
-
-	if q == nil {
-		errs = append(errs, "cannot specify a nil queue")
-	}
+	catcher := grip.NewBasicCatcher()
+	catcher.NewWhen(targetNum <= 0, "cannot specify a target number of jobs less than 1")
+	catcher.NewWhen(size <= 0, "cannot specify a pool size less than 1")
+	catcher.NewWhen(period < time.Millisecond, "cannot specify a scheduling period interval less than a millisecond")
+	catcher.NewWhen(q == nil, "cannot specify a nil queue")
```
Comment (julianedwards) [inline pool/rate_limiting_average.go:29]: `must specify a queue`

File: pool/rate_limiting_average.go [hunk 2]
```diff
@@ -86,7 +73,7 @@ func (p *ewmaRateLimiting) getNextTime(dur time.Duration) time.Duration {
 	tasksPerPeriod := p.period / averageRuntime
 
 	// the capacity of the pool is the size of the pool and the
```
Comment (julianedwards) [inline pool/rate_limiting_average.go:75]: `The`

File: pool/rate_limiting_average.go [hunk 3]
```diff
@@ -86,7 +73,7 @@ func (p *ewmaRateLimiting) getNextTime(dur time.Duration) time.Duration {
...
-	// target number of tasks
+	// target number of jobs
```
Comment (julianedwards) [inline pool/rate_limiting_average.go:76]: `... jobs.`

File: queue/group_remote_mongo.go [hunk 4]
```diff
@@ -59,11 +59,11 @@ type MongoDBQueueGroupOptions struct {
...
-	catcher.NewWhen(o.TTL < 0, "TTL must be greater than or equal to 0")
-	catcher.NewWhen(o.TTL > 0 && o.TTL < time.Second, "TTL cannot be less than 1 second, unless it is 0")
-	catcher.NewWhen(o.PruneFrequency < 0, "prune frequency must be greater than or equal to 0")
-	catcher.NewWhen(o.PruneFrequency > 0 && o.TTL < time.Second, "prune frequency cannot be less than 1 second, unless it is 0")
-	catcher.NewWhen((o.TTL == 0 && o.PruneFrequency != 0) || (o.TTL != 0 && o.PruneFrequency == 0), "ttl and prune frequency must both be 0 or both be not 0")
+	catcher.NewWhen(o.TTL < 0, "TTL cannot be negative")
+	catcher.NewWhen(o.TTL > 0 && o.TTL < time.Second, "TTL cannot be less than 1 second, unless it is not set")
+	catcher.NewWhen(o.PruneFrequency < 0, "prune frequency cannot be negative")
+	catcher.NewWhen(o.PruneFrequency > 0 && o.TTL < time.Second, "prune frequency cannot be less than 1 second, unless it is not set")
+	catcher.NewWhen((o.TTL == 0 && o.PruneFrequency != 0) || (o.TTL != 0 && o.PruneFrequency == 0), "TTL and prune frequency must both be set or both be not set")
```
Comment (julianedwards) [inline queue/group_remote_mongo.go:66]: `must specify TTL and prune frequency or neither`

File: queue/group_remote_mongo.go [hunk 5]
```diff
@@ -59,11 +59,11 @@ type MongoDBQueueGroupOptions struct {
 
 func (o MongoDBQueueGroupOptions) validate() error {
 	catcher := grip.NewBasicCatcher()
-	catcher.NewWhen(o.TTL < 0, "TTL must be greater than or equal to 0")
-	catcher.NewWhen(o.TTL > 0 && o.TTL < time.Second, "TTL cannot be less than 1 second, unless it is 0")
-	catcher.NewWhen(o.PruneFrequency < 0, "prune frequency must be greater than or equal to 0")
-	catcher.NewWhen(o.PruneFrequency > 0 && o.TTL < time.Second, "prune frequency cannot be less than 1 second, unless it is 0")
-	catcher.NewWhen((o.TTL == 0 && o.PruneFrequency != 0) || (o.TTL != 0 && o.PruneFrequency == 0), "ttl and prune frequency must both be 0 or both be not 0")
+	catcher.NewWhen(o.TTL < 0, "TTL cannot be negative")
+	catcher.NewWhen(o.TTL > 0 && o.TTL < time.Second, "TTL cannot be less than 1 second, unless it is not set")
```
Comment (julianedwards) [inline queue/group_remote_mongo.go:63]: `cannot specify a TTL less than a second, unless unspecified`

File: queue/group_remote_mongo.go [hunk 6]
```diff
@@ -59,11 +59,11 @@ type MongoDBQueueGroupOptions struct {
...
+	catcher.NewWhen(o.PruneFrequency < 0, "prune frequency cannot be negative")
+	catcher.NewWhen(o.PruneFrequency > 0 && o.TTL < time.Second, "prune frequency cannot be less than 1 second, unless it is not set")
```
Comment (julianedwards) [inline queue/group_remote_mongo.go:65]: `cannot specify a prune frequency less than a second, unless unspecified`

File: queue/ordered.go [hunk 7]
```diff
@@ -109,11 +109,11 @@ func (q *depGraphOrderedLocal) Put(ctx context.Context, j amboy.Job) error {
 	defer q.mutex.Unlock()
 
 	if q.started {
-		return errors.Errorf("cannot add %s because ordered task dispatching has begun", name)
+		return errors.Errorf("cannot add job '%s' because ordered task dispatching has begun", name)
 	}
 
 	if _, ok := q.tasks.m[name]; ok {
-		return amboy.NewDuplicateJobErrorf("cannot add %s because duplicate job ids are not allowed", name)
+		return amboy.NewDuplicateJobErrorf("cannot add job '%s' because duplicate job ids are not allowed", name)
```
Comment (julianedwards) [inline queue/ordered.go:116]: `IDs`

File: queue/shuffled.go [hunk 8]
```diff
@@ -102,7 +102,7 @@ func (q *shuffledLocal) Put(ctx context.Context, j amboy.Job) error {
 	id := j.ID()
 
 	if !q.Info().Started {
-		return errors.Errorf("cannot put job %s; queue not started", id)
+		return errors.New("cannot enqueue jobs when queue not active")
```
Comment (julianedwards) [inline queue/shuffled.go:105]: `... when queue is not active`

**Review by julianedwards — CHANGES_REQUESTED** [review]

File: management/interface.go [hunk 9]
```diff
@@ -64,7 +64,7 @@ func (t StatusFilter) Validate() error {
 	case InProgress, Pending, Stale, Completed, Retrying, All:
 		return nil
 	default:
-		return errors.Errorf("%s is not a valid status filter type", t)
+		return errors.Errorf("'%s' is not a valid status filter type", t)
```
[the remark this answers is not in the document]
Comment (Kimchelly) [reply management/interface.go:67]: Done.

File: periodic.go [hunk 10]
```diff
@@ -171,7 +171,7 @@ func scheduleOp(ctx context.Context, q Queue, op QueueOperation, conf QueueOpera
 		return nil
 	}
 
-	if err := errors.Wrap(op(ctx, q), "problem encountered during periodic job scheduling"); err != nil {
+	if err := errors.Wrap(op(ctx, q), "periodic job operation"); err != nil {
```
[the remark this answers is not in the document]
Comment (Kimchelly) [reply periodic.go:174]: This is running the op. I fixed it.

File: pool/rate_limiting_average.go (hunk 1 above)
Comment (Kimchelly) [reply pool/rate_limiting_average.go:29]: Done.

File: pool/rate_limiting_average.go (hunk 2 above)
Comment (Kimchelly) [reply pool/rate_limiting_average.go:75]: Fixed this comment and the other comments in `getNextTime`.

File: pool/rate_limiting_average.go (hunk 3 above)
Comment (Kimchelly) [reply pool/rate_limiting_average.go:76]: Fixed.

File: queue/group_remote_mongo.go (hunk 5 above)
Comment (Kimchelly) [reply queue/group_remote_mongo.go:63]: I rephrased it into an if-then.

File: queue/group_remote_mongo.go (hunk 6 above)
Comment (Kimchelly) [reply queue/group_remote_mongo.go:65]: Same as above.

File: queue/group_remote_mongo.go (hunk 4 above)
Comment (Kimchelly) [reply queue/group_remote_mongo.go:66]: Done.

File: queue/ordered.go (hunk 7 above)
Comment (Kimchelly) [reply queue/ordered.go:116]: Done.

File: queue/shuffled.go (hunk 8 above)
Comment (Kimchelly) [reply queue/shuffled.go:105]: Done.

File: periodic.go [hunk 11]
```diff
@@ -20,8 +20,8 @@ type QueueOperation func(context.Context, Queue) error
 // QueueOperationConfig describes the behavior of the periodic
 // interval schedulers.
 //
-// The theshold, if ResepectThreshold is set, causes the periodic
-// scheduler to noop if there are more than that many pending jobs.
+// The theshold, if RespectThreshold is set, causes the periodic
```
Comment (julianedwards) [inline periodic.go:23]: `threshold`

**Review by julianedwards — APPROVED** [review]
LGTM % mispelling.

**Pull request merged**